dsp/modules/aws_models.py

```diff
@@ -0,0 +1,302 @@
+            base_args[k] = v
+
+        n, base_args = self.aws_provider.sanitize_kwargs(base_args)
+
+        query_args: dict[str, str | float] = {}
+        if isinstance(self.aws_provider, Bedrock):
+            query_args["prompt"] = prompt
+        elif isinstance(self.aws_provider, Sagemaker):
+            query_args["parameters"] = base_args
+            query_args["inputs"] = prompt
+        else:
+            raise ValueError("Error - provider not recognized")
+
+        return (n, query_args)
+
+    def _call_model(self, body: str) -> str:
+        response = self.aws_provider.call_model(
+            model_id=self._model_name,
+            body=body,
+        )
+        if isinstance(self.aws_provider, Bedrock):
+            response_body = json.loads(response["body"].read())
+            completion = response_body["outputs"][0]["text"]
+        elif isinstance(self.aws_provider, Sagemaker):
+            response_body = json.loads(response["Body"].read())
+            completion = response_body[0]["generated_text"]
+        else:
+            raise ValueError("Error - provider not recognized")
+
+        completion = completion.split(self.kwargs["stop"])[0]
+        return completion
+
+
+class AWSAnthropic(AWSModel):
+    """Anthropic family of models."""
+
+    def __init__(
+        self,
+        aws_provider: AWSProvider,
+        model: str,
+        max_context_size: int = 200000,
+        max_new_tokens: int = 1500,
+        **kwargs,
+    ) -> None:
+        """NOTE: Configure your AWS credentials with the AWS CLI before using this model!"""
+        super().__init__(
+            model=model,
+            max_context_size=max_context_size,
+            max_new_tokens=max_new_tokens,
+            **kwargs,
+        )
+        self.aws_provider = aws_provider
+        self.provider = aws_provider.get_provider_name()
+
+        for k, v in kwargs.items():
+            self.kwargs[k] = v
+
+    def _create_body(self, prompt: str, **kwargs) -> tuple[int, dict[str, str | float]]:
+        base_args: dict[str, Any] = self.kwargs
+        for k, v in kwargs.items():
+            base_args[k] = v
+
+        n, query_args = self.aws_provider.sanitize_kwargs(base_args)
+
+        # Anthropic models do not support the following parameters
+        query_args.pop("frequency_penalty", None)
+        query_args.pop("num_generations", None)
+        query_args.pop("presence_penalty", None)
+        query_args.pop("model", None)
+
+        # we are using the Claude messages API
+        # https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-anthropic-claude-messages.html
+        query_args["messages"] = [
+            {
+                "role": "user",
+                "content": [
+                    {
+                        "type": "text",
+                        "text": prompt,
+                    },
+                ],
+            },
+        ]
+        query_args["anthropic_version"] = "bedrock-2023-05-31"
```

Comment:
should this be set through the kwargs?

Reply:
Good catch! I moved this to the constructor instead and made it conditional to Bedrock. Users can override. 
Ideally, we can put such params in a config file. Do we have support config files?